### ISSUE TITLE
Azure Cosmos DB output binding does not stored documents as expected

### ISSUE THREAD

**Issue by paolosalvatori** [Azure/azure-functions-python-worker#122]
I created a sample using Azure Functions in Python. The function works as expected both locally and on Azure. The function uses a HTTP trigger to receive a request and return a response. It also uses an Azure Cosmos DB outbound binding to store results to Azure Cosmos DB, but no document is stored in my Azure Cosmos DB collection.

**``__init__.py``**

```python
import logging
import azure.functions as func
...
def main(req: func.HttpRequest, doc: func.Out[func.Document]) -> func.HttpResponse:
...
        # create json file as a result
        result = json.dumps({'id':url, 'url': url, 'links': links}, indent=4, sort_keys=False)

        # store json document to Azure Cosmos DB
        # this line of code does not store the doc as expected to Azure Cosmos DB
        # However, it does not raise any exception
        doc.set(func.Document.from_json(result))
        
        # return json document
        return func.HttpResponse(result)
    else:
        return func.HttpResponse(
             "Please pass a valid url on the query string or in the request body",
             status_code=400
        )
```

**``function.json``**

```json
{
  "scriptFile": "__init__.py",
  "disabled": false,
  "bindings": [
    {
      "authLevel": "function",
      "type": "httpTrigger",
      "direction": "in",
      "name": "req"
    },
	{
      "direction": "out",
      "type": "cosmosDB",
      "name": "doc",
      "databaseName": "linksdb",
      "collectionName": "items",
	  "createIfNotExists": true,
	  "partitionKey": "/url",
	  "connection": "AzureCosmosDbConnectionString"
    },
    {
      "type": "http",
      "direction": "out",
      "name": "$return"
    }
  ]
}
```

**``local.settings.json``**

```json
{
  "IsEncrypted": false,
  "Values": {
    "FUNCTIONS_WORKER_RUNTIME": "python",
    "AzureWebJobsStorage": "...",
	"AzureCosmosDbConnectionString": "AccountEndpoint=https://xxxxxx.documents.azure.com:443/;AccountKey=xxxx==;"
  }
}
```

Am I doing anything wrong or is the Azure Cosmos DB output binding

**Comment by elprans** [comment]
@paolosalvatori Please attach the full log produced by the host.  

We have an integration test that works, so this is likely a local host/configuration issue.

**Comment by asavaritayal** [comment]
I think you may be missing the step to install the Cosmos Db extension for functions: 

```bash
func extensions install --package Microsoft.Azure.WebJobs.Extensions.CosmosDB --version 3.0.0-beta7
```
I'll add this to the docs as well.

**Comment by paolosalvatori** [comment]
Thanks @asavaritayal! I proceeded as follows:

1. Ran the following command to add the Azure Cosmos DB extension as you suggested:

func extensions install --package Microsoft.Azure.WebJobs.Extensions.CosmosDB --version 3.0.0-beta7

2. Added the AzureWebJobsCosmosDBConnectionString with the Azure Cosmos DB connection string to the local.settings.json

{
  "IsEncrypted": false,
  "Values": {
    "FUNCTIONS_WORKER_RUNTIME": "python",
    "AzureWebJobsStorage": XXXX==",
    "AzureWebJobsCosmosDBConnectionString": XXXX==;"
  }
}

3. Added the "leaseCollectionName": "leases" to function.json:

{
  "scriptFile": "__init__.py",
  "disabled": false,
  "bindings": [
    {
      "authLevel": "anonymous",
      "type": "httpTrigger",
      "direction": "in",
      "name": "req"
    },
	{
      "direction": "out",
      "type": "cosmosDB",
      "name": "doc",
      "databaseName": "linksdb",
      "collectionName": "items",
      "leaseCollectionName": "leases",
      "createIfNotExists": true,
      "partitionKey": "/url"
    },
    {
      "type": "http",
      "direction": "out",
      "name": "$return"
    }
  ]
}

4. Successfully tested the application locally (it stored the document to Azure Cosmos DB as expected)
5. Added the AzureWebJobsCosmosDBConnectionString to the app settings of my Azure Function on Azure
6. Published the function app to Azure using func azure functionapp publish PythonDockerApp
7. Successfully tested the app on Azure
